### PR TITLE
Start handling errors on stream/astream/astream_log

### PR DESCRIPTION
**Pull request by eyurtsev**
This PR introduces error handling for stream/astream/astream_log endpoints.

The error handling is incomplete as client side errors aren't handled yet.

1) When an exception is encountered while streaming the server sents an error event in the stream with data:

- status code: int
- message: str

2) Client code is updated to handle this

In addition this PR, proposes that the client with log an error instead of
raising when encountering an unsupported event. This will allow us to ship
additional events without breaking older clients. Useful for sending over
callback events.

### CODE REVIEW

File: langserve/server.py [hunk 1]
```diff
@@ -464,31 +477,44 @@ async def stream_log(
...
+                    "event": "error",
+                    # Do not expose the error message to the client since
+                    # the message may contain sensitive information.
+                    # We'll add client side errors for validation as well.
+                    "data": json.dumps(
+                        {"status_code": 500, "message": "Internal Server Error"}
+                    ),
                 }
-            yield {"event": "end"}
+                yield {"event": "end"}
```
Comment (efriis) [inline langserve/server.py:516]: Should `error` imply an `end`?

Comment (eyurtsev) [reply langserve/server.py:516]: Wasn't sure about that -- we can have it imply end or always send an end event. Any preferences? 

b/c older clients will break anyway, neither one is backwards compatible

**Review by efriis — APPROVED** [review]
good with the yield end removal!

**Comment by rere950303** [comment]
@eyurtsev thanks for your work

When using an error event as a response, the status_code and message are currently hardcoded as follows. However, in some cases, there may be requirements to show different results to the client by referring to the status_code of the error response. Therefore, I hope that it will be developed so that status_code and message can be customized. Of course, sensitive information can be exposed like the annotation written, but there are many ways not to expose sensitive information in a way such as defining status_code, message, etc. through a custom exception class. If development is impossible, is there a way for me to customize the currently developed code?

**Pull request merged**